python-manual/modules/ROOT/pages/result-summary.adoc

```diff
@@ -0,0 +1,148 @@
+
+[source, python]
+----
+def create_people(tx):
+    result = tx.run("""
+        UNWIND ["Alice", "Bob"] AS name
+        MERGE (p:Person {name: name})
+    """)
+    return result.consume()
+
+with driver.session(database="neo4j") as session:
+    result_summary = session.execute_write(create_people)
+----
+
+
+## Query counters
+
+The property `ResultSummary.counters` contains counters for the operations that a query triggered (as a link:{neo4j-docs-base-uri}/api/python-driver/current/api.html#summarycounters[`SummaryCounters`] object).
+
+.Insert some data and display the query counters
+[source, python]
+----
+summary = driver.execute_query("""
+    MERGE (p:Person {name: $name})
+    MERGE (p)-[:KNOWS]->(:Person {name: $friend})
+    """, name="Mark", friend="Bob",
+    database_="neo4j",
+).summary
+print(summary.counters)
+"""
+{'_contains_updates': True, 'labels_added': 2, 'relationships_created': 1,
+ 'nodes_created': 2, 'properties_set': 2}
+"""
+----
+
+There are two additional boolean properties which act as meta-counters:
+
+- `contains_updates` -- whether the query triggered any write operation on the database on which it ran
+- `contains_system_updates` -- whether the query updated the `system` database
+
+
+## Notifications
+
+The property `ResultSummary.summary_notifications` contains a list of link:{neo4j-docs-base-uri}/status-codes/current/notifications[notifications coming from the server], if any were raised by the execution of the query.
+These include recommendations for performance improvements, warnings about the usage of deprecated features, and other hints about sub-optimal usage of Neo4j.
+Each notification comes as a link:{neo4j-docs-base-uri}/api/python-driver/current/api.html#neo4j.SummaryNotification[`SummaryNotification`] object.
+
+.An unbounded shortest path raises a performance notification
+[source, python, role=nocollapse]
+----
+records, summary, keys = driver.execute_query("""
+    MATCH p=shortestPath((:Person {name: 'Alice'})-[*]->(:Person {name: 'Bob'}))
+    RETURN p
+    """, database_="neo4j"
+)
+print(summary.summary_notifications)
+"""
+[SummaryNotification(
+    title='The provided pattern is unbounded, consider adding an upper limit to the number of node hops.', code='Neo.ClientNotification.Statement.UnboundedVariableLengthPattern',
+    description='Using shortest path with an unbounded pattern will likely result in long execution times. It is recommended to use an upper limit to the number of node hops in your pattern.',
+    severity_level=<NotificationSeverity.INFORMATION: 'INFORMATION'>,
+    category=<NotificationCategory.PERFORMANCE: 'PERFORMANCE'>,
+    raw_severity_level='INFORMATION',
+    raw_category='PERFORMANCE',
+    position=SummaryNotificationPosition(line=1, column=22, offset=21)
+)]
+"""
+----
+
+
+### Filtering notifications
+
+By default, the server analyses each query for all categories and severity of notifications.
```

Comment:
Ooh. I've always been wondering if we prefer our docs in uk-en or us-en.

Reply:
Yeah. I'll pretend I picked one _on purpose_.